ts/Series/Treegraph/TreegraphSeries.ts

```diff
@@ -0,0 +1,831 @@
+import TreegraphNode from './TreegraphNode.js';
+import TreegraphPoint from './TreegraphPoint.js';
+import TU from '../TreeUtilities.js';
+import PathUtilities from '../PathUtilities.js';
+const { prototype: { symbols } } = SVGRenderer;
+const { getLevelOptions } = TU;
+const {
+    seriesTypes: {
+        treemap: TreemapSeries,
+        column: ColumnSeries
+    }
+} = SeriesRegistry;
+
+import H from '../../Core/Globals.js';
+import U from '../../Core/Utilities.js';
+const { merge, pick, addEvent, isArray } = U;
+
+import './TreegraphLayout.js';
+import TreegraphLink from './TreegraphLink.js';
+import { Palette } from '../../Core/Color/Palettes.js';
+
+interface LayoutModifiers {
+    ax: number;
+    bx: number;
+    ay: number;
+    by: number;
+}
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The Treegraph series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.treegraph
+ *
+ * @augments Highcharts.Series
+ */
+class TreegraphSeries extends TreemapSeries {
+    /* *
+     *
+     *  Static Properties
+     *
+     * */
+
+    /**
+     * A treegraph series is a diagram, which shows a relation between ancestors
+     * and descendants with a clear parent - child relation.
+     * The best examples of the dataStructures, which best reflect this chart
+     * are e.g. genealogy tree or directory structure.
+     *
+     * @sample highcharts/demo/treegraph-chart
+     *          Treegraph Chart
+     *
+     * @extends      plotOptions.treemap
+     * @excluding    layoutAlgorithm, dashStyle, linecap, lineWidth,
+     *               negativeColor, threshold, zones, zoneAxis, colorAxis,
+     *               colorKey, compare, dataGrouping, endAgle, gapSize, gapUnit,
+     *               ignoreHiddenPoint, innerSize, joinBy, legendType, linecap,
+     *               minSize, navigatorOptions, pointRange, allowTraversingTree,
+     *               alternateStartingDirection, borderRadius, breadcrumbs,
+     *               interactByLeaf, layoutStartingDirection, levelIsConstant,
+     *               lineWidth, negativeColor, nodes, sortIndex, zoneAxis,
+     *               zones
+     *
+     * @since        next
+     * @product      highcharts
+     * @requires     modules/treemap.js
+     * @requires     modules/treegraph.js
+     * @optionparent plotOptions.treegraph
+     */
+    public static defaultOptions: TreegraphSeriesOptions = merge(
+        TreemapSeries.defaultOptions,
+        {
+            /**
+             * The option to choose the layout alogrythm, which should be used
+             * to calculate the positions of the nodes.
+             *
+             *
+             * @sample highcharts/demo/treegraph-chart
+             *         Treegraph chart
+             *
+             * @type {'Walker'}
+             * @since next
+             * @default 'Walker'
+             * @product highcharts
+             */
+            layout: 'Walker',
```

Comment:
For now, I would ignore this option since there's no other possible value. We can add this option later, when more algorithms will be added